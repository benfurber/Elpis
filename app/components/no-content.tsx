import React, { Component } from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import { BackgroundContainer, ErrorMessage, Icon } from "components";
import { NavigationType } from "interfaces";
import { labels } from "labels";
import { layout } from "styles";

interface Props {
  navigation: NavigationType;
}

class NoContent extends Component<Props> {
  render() {
    const { navigation } = this.props;

    const goBack = () => navigation.pop();
    const message = labels.errors.noContent;

    return (
      <BackgroundContainer>
        <TouchableOpacity style={styles.back} onPress={() => goBack()}>
          <Icon name="angle-double-left" style={styles.icon} />
          <Text>{labels.back}</Text>
        </TouchableOpacity>

        <ErrorMessage error={{ message }} />
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  back: {
    paddingTop: layout.spacingL,
    paddingHorizontal: layout.spacingL,
    flexDirection: "row",
  },
  icon: {
    paddingRight: layout.spacingS,
  },
});

export { NoContent };
