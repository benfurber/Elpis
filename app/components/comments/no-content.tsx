import React, { Component } from "react";
import { Image, StyleSheet, View } from "react-native";

import { Title } from "components";
import { layout, typography } from "styles";

const noCommentsBoxPath = "../../assets/images/no-comments-box.png";

interface Props {
  text: string;
}

class NoContent extends Component<Props> {
  render() {
    return (
      <View style={styles.container}>
        <Title style={styles.title} text={this.props.text} />
        <Image style={styles.image} source={require(noCommentsBoxPath)} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: layout.spacing,
    paddingTop: layout.spacingL,
    paddingBottom: layout.spacingXL,
  },
  image: {
    alignSelf: "center",
    top: -30,
    zIndex: -1,
  },
  title: {
    alignContent: "center",
    alignSelf: "center",
    fontSize: typography.fontSizeXL,
    textAlign: "center",
  },
});

export { NoContent };
