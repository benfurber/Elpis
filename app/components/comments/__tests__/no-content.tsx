import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { NoContent } from "../no-content";

describe("Comments<NoContent>", () => {
  it("renders correctly", () => {
    const component = shallow(<NoContent text={"No content to show!"} />);

    expect(component).toMatchSnapshot();
  });
});
