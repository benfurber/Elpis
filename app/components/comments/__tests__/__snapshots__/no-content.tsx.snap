// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments<NoContent> renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <Title
    text="No content to show!"
  />
</View>
`;
